Agent/Linux/ubuntu1804-README.md

```diff
@@ -1,44 +1,64 @@
+2. Generate a [Personal Access Token (PAT)](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops#create-personal-access-tokens-to-authenticate-access) for your Azure DevOps Organization. When generating the Personal Access Token (PAT), assign the following scopes:
 
->a. Agent Pools - Read & Manage
->b. Deployment Groups - Read & Manage
+* Agent Pools - Read & Manage
+* Deployment Groups - Read & Manage
 
-3. Create a service principle and add it to your subscription as Owner. This is the subscription that will be used to host the image and Agent VM.
+3. Create a [Service Principal](https://docs.microsoft.com/en-us/powershell/azure/create-azure-service-principal-azureps?view=azps-2.4.0).
 
-4. Add the service principle to the subscription that will host the image and agent VMs with Owner role.
+4. Add [RBAC Assignment](https://docs.microsoft.com/en-us/azure/role-based-access-control/role-assignments-portal#add-a-role-assignment) to the service principal created in previous step by adding the service prinicpal as Owner to the subscription that will host the image and agent VMs with Owner role.
 
 ## Build a Packer Image
 
-1. Build a Docker image and a container from the Docker image by mounting `Agent` folder as a volume in your Docker container.
+1. Build a Docker image.
+
+```
+docker build --rm -f "dockerfile" -t vdc:latest .
+```
+
+> Note: Do not miss the `.` at the end when running the above command.
+
+2. Create a container from the Docker image built in the previous step. Mount the `Agent` folder as a volume in your Docker container when creating the container.
 
 ```
 docker run -it -v <path-to-vdc-folder>/Agent:/usr/app/src/Agent vdc:latest
 ```
 
-2. Update the run.sh file with the service principle and subscription information.
+2. Update the build.sh file with the service principal's client id, client secret, subscription id and tenant id.
 
-3. Run the command below from PowerShell terminal to build an VM Image
+3. Run the command below from PowerShell terminal to build a VM Image
 
 ```
 bash -c "./build.sh"
 ```
 
+> VM Image will contain all the required tools, SDKs and CLIs to run the VDC toolkit.
+
 ## Create a VM from Packer Image
 
-1. Run the command below from PowerShell terminal to create an instance from the VM Image
+1. Run the command below from PowerShell terminal to create a VM from the Image
 
 ```
 bash -c "./vsts-agent-create.sh <organization-url> <personal-access-token> <vsts-pool-name> <agent-vm-name>"
 ```
 
+> VM will be created in the same resource group, subscription and tenant as the Image created by Packer. The name of this resource group can be found in the build.sh and vsts-agent-create.sh bash scripts.
+> VM will have following attributes:
+> * Created from the Image built by Packer
+> * No Public IP
+> * No NSG rule to allow SSH connections
+> * Explicit NSG rule to block all SSH connections
```

Comment:
Add a comment to this line with something like this: 

If you need to allow SSH connectivity from your computer, enable an SSH connection with a specific IP source.

Reply:
Done.